### PR TITLE
 Prompt for terms of service on identity server changes

### PR DESCRIPTION
**Pull request by turt2live**
**Requires https://github.com/matrix-org/matrix-react-sdk/pull/3316**
Part of https://github.com/vector-im/riot-web/issues/10539

See https://github.com/matrix-org/matrix-react-sdk/pull/3316 for screenshots (they're the same, just using "identity server" instead of "integration manager").

### CODE REVIEW

File: src/components/views/settings/SetIdServer.js [hunk 1]
```diff
@@ -127,28 +130,68 @@ export default class SetIdServer extends React.Component {
         return !!this.state.idServer && !this.state.busy;
     };
 
+    _continueTerms = (fullUrl) => {
```
Comment (jryans) [inline src/components/views/settings/SetIdServer.js]: `_continueTerms` seems like a confusing name for something that saves the new server to storage... Maybe this should become:

```suggestion
    _saveIdServer = (fullUrl) => {
```

File: src/components/views/settings/SetIdServer.js [hunk 2]
```diff
@@ -127,28 +130,68 @@ export default class SetIdServer extends React.Component {
...
+        MatrixClientPeg.get().setIdentityServerUrl(fullUrl);
+        localStorage.removeItem("mx_is_access_token");
+        localStorage.setItem("mx_is_url", fullUrl);
+        dis.dispatch({action: 'id_server_changed'});
+        this.setState({idServer: '', busy: false, error: null});
+    };
+
     _saveIdServer = async (e) => {
```
Comment (jryans) [inline src/components/views/settings/SetIdServer.js]: ...and this should change to something like:

```suggestion
    _checkIdServer = async (e) => {
```

File: src/IdentityAuthClient.js [hunk 3]
```diff
@@ -14,15 +14,50 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { SERVICE_TYPES } from 'matrix-js-sdk';
+import Matrix, { SERVICE_TYPES } from 'matrix-js-sdk';
```
Comment (jryans) [inline src/IdentityAuthClient.js]: Consider importing only `createClient`.

File: src/components/views/settings/SetIdServer.js (hunk 1 above)
Comment (turt2live) [reply src/components/views/settings/SetIdServer.js]: indeed - will apply both suggestions manually.

**Pull request merged**